crates/net/network/src/discovery.rs

```diff
@@ -41,6 +45,8 @@ pub struct Discovery {
     _dns_disc_service: Option<JoinHandle<()>>,
     /// Events buffered until polled.
     queued_events: VecDeque<DiscoveryEvent>,
+
+    discovery_manager_tx: UnboundedSender<DiscoveryEvent>,
```

Comment:
there can be multiple listeners
what we need is a list of senders just like this:

https://github.com/paradigmxyz/reth/blob/b07dfb55259db022455e267900a02e599869b78b/crates/net/discv4/src/lib.rs#L392-L393

and 

https://github.com/paradigmxyz/reth/blob/b07dfb55259db022455e267900a02e599869b78b/crates/net/discv4/src/lib.rs#L686-L695